swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/events/ResourceStateManager.kt

```diff
@@ -176,19 +176,19 @@ class ResourceStateManager(
       when {
         equals("serverGroup", true) -> tagAsg(resource, workConfiguration)
         equals("image", true) -> tagImage(resource)
-        else -> log.error("Failed to tag resource ${resource.uniqueId()}")
+        else -> log.error("Failed to tag resource ${resource.uniqueId()} with aws expiration_time tag ")
       }
     }
     taskTrackingRepository.add(
-      taskId.toString(),
+      taskId!!.toString(),
```

Comment:
The `!!` asserts that this will never be `null`. Instead, you need to correctly handle if the `taskId` is `null` by not adding it to the `taskTrackingRepository`.

Reply:
done